lib/OpenQA/Client/Archive.pm

```diff
@@ -0,0 +1,198 @@
+package OpenQA::Client::Archive;
+use strict;
+use warnings;
+use Data::Dump;
+
+use JSON;
+use Mojo::URL;
+use Mojo::File 'path';
+
+our $client;
+our %options;
+our $job;
+our $path;
+our $url;
+
+sub run {
+    #We're only taking one argument
+    ($client, %options) = @_;
```

Comment:
the number of global variables is scary. From what I can see the only entrance into this class is ::run - better make that a real class then.
```
- Archive::run($client, %options)
+ Archive->new($client,%options)->run()
```

and then use $self to pass around the features

Reply:
Done :)